### PR TITLE
CHORE: Author Pages

### PR DESCRIPTION
**Pull request by jamesbiggs**
## About these changes

This PR is for the creation of the "author" page type.

**Features:**
- Create an `AuthorPage`, which includes a profile image, a role description, and a summary/"about the author" section.
- These pages can then be used on `FocusedArticlePage`, to tag a page with an author - similar to how we tag with topic/time periods.
- This then adds the authors icon/info to the page, and relates backwards to the author page, allowing us to display any authored pages on their author page.

**Fixes:**
I noticed that the `slug` field wasn't working. This is due to us overwriting the default `help_text` that we inherit from the `Page` type, into our `BasePage`. I added in a fix for this to update the structure of the fields that we have, so that the `slug` field works properly. Currently, it's taking the title and just copying it - it should be "slugifying" it. This fix should do this for us here.

**Note:**
This PR is in draft and will be completed when:
- I have made some test cases for these pages
- The front-end is completed (currently just a rough example)

## How to check these changes

Create author index page, create author page under this. Tag a `FocusedArticlePage` with this author, and see that the author info is on the page, and that the author page has the "tagged" page(s).

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/authors/author_index_page.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            {% for author_page in page.author_pages %}
+                
+                <li class="col-sm-12 col-md-6 col-lg-4 mb-3">
+                    <div class="card-group-secondary-nav">
+                        <a
+                            href="{{ author_page.url }}"
+                            class="card-group-secondary-nav__image-link"
+                            data-card-type="card-group-secondary-nav"
+                            data-link-type="Card image"
+                            data-card-title="{{ author_page.title }}"
```
Comment (ahosgood) [inline templates/authors/author_index_page.html:22]: Do we use these `data-` attributes for tracking or something? I've seen them in quite a few places now.

Comment (jamesbiggs) [reply templates/authors/author_index_page.html:22]: Yeah it's for tracking with GA4 😃 I've removed some of them from the bits I've copied from other elements just so I wasn't constantly changing the values of it 😅

**Comment by jamesbiggs** [comment]
@ahosgood @JohnHeeryTNA This PR is ready whenever the FE is ready! 😃

The only thing that may need changing is the author index page - I personally think we should have it but others disagree, even if it's just for semantics and to group everything together in the CMS.

File: etna/articles/models.py [hunk 2]
```diff
@@ -321,10 +323,6 @@ class FocusedArticlePage(
     The FocusedArticlePage model.
     """
 
-    author = models.CharField(
-        max_length=100, blank=True, null=True, help_text="The author of this article."
-    )
```
Comment (ababic) [inline etna/articles/models.py:326]: @jamesbiggs Something like this should do:

```
author = models.ForeignKey("authors.AuthorPage", blank=True, null=True, related_name="focussed_articles")
```

Comment (ababic) [reply etna/articles/models.py:326]: With the above, you should be able to do this from the author page to list the articles:
```
self.focussed_articles.live().public()
```

File: etna/articles/models.py [hunk 3]
```diff
@@ -343,7 +341,6 @@ class Meta:
         BasePageWithIntro.content_panels
         + HeroImageMixin.content_panels
         + [
-            FieldPanel("author"),
```
Comment (ababic) [inline etna/articles/models.py:346]: If you pop this back in, Wagtail should automatically give you a PageChooser here to select one.

File: etna/articles/models.py [hunk 4]
```diff
@@ -376,6 +374,7 @@ class Meta:
             index.SearchField("body"),
             index.SearchField("topic_names", boost=1),
             index.SearchField("time_period_names", boost=1),
+            index.SearchField("author_name", boost=1),
```
Comment (ababic) [inline etna/articles/models.py:377]: You might want to consider using [index.RelatedFields](https://docs.wagtail.org/en/stable/topics/search/indexing.html#index-relatedfields) here instead.

File: etna/authors/models.py [hunk 5]
```diff
@@ -28,24 +49,90 @@ class Author(models.Model):
...
+    def authored_focused_articles(self):
+        from etna.articles.models import FocusedArticlePage
+
+        return (
+            FocusedArticlePage.objects.live()
+            .public()
+            .filter(pk__in=self.related_page_pks)
+            .order_by("-first_published_at")
+            .select_related("teaser_image")
+        )
```
Comment (ababic) [inline etna/authors/models.py:75]:
```suggestion
    def authored_focused_articles(self):
        return (
            self.focussed_articles.live()
            .public()
            .order_by("-first_published_at")
            .select_related("teaser_image")
        )
```

File: etna/authors/models.py [hunk 6]
```diff
@@ -28,24 +49,90 @@ class Author(models.Model):
...
+        """
+        Returns a list of ids of pages that have used the `AuthorTag` inline
+        to indicate a relationship with this author. The values are ordered by
+        when the page was first published ('more recently added' pages take presendence)
+        """
+        return tuple(
+            self.author_pages.values_list("page_id", flat=True).order_by(
+                "-page__first_published_at"
+            )
+        )
```
Comment (ababic) [inline etna/authors/models.py:88]: You should be able to drop this method

File: templates/articles/focused_article_page.html [hunk 7]
```diff
@@ -14,11 +15,11 @@
     {% else %}
         {% include 'includes/generic-intro.html' with breadcrumb=True title=page.title %}
     {% endif %}    
-    {% if page.author or page.first_published_at %}
+    {% if page.author_name or page.first_published_at %}
         <div class="generic-intro__meta">
             <ul class="generic-intro__meta-list container">
-                {% if page.author %}
-                    <li class="generic-intro__meta-item">By {{ page.author }}</li>                    
+                {% if page.author_name %}
```
Comment (ababic) [inline templates/articles/focused_article_page.html:21]: Is this second conditional needed here?

**Comment by jamesbiggs** [comment]
@ababic I've made the changes 😃

**Review by Firdaussi — APPROVED** [review]
I'm happy with the changes made and the issues addressed by Andy.

**Comment by LeanneEmans** [comment]
@jamesbiggs if this PR has been approved by Ian, can it be merged now please?

**Pull request merged**